integration/spark/src/main/scala/org/apache/spark/sql/CarbonScan.scala

```diff
@@ -151,8 +153,15 @@ case class CarbonScan(
 
         override def next(): InternalRow = {
           val value = iter.next
+          val converted = value.map {
```

Comment:
understood, I will wait for the performance check in cluster environment before reworking on this